Frameworks/CoreText/CTFramesetter.mm

```diff
@@ -87,18 +88,22 @@ CTFrameRef CTFramesetterCreateFrame(CTFramesetterRef framesetter, CFRange string
     _CGPathGetBoundingBoxInternal(path, &frameSize);
 
     CGSize sizeOut;
-    _CTFrame* ret = _createFrame((_CTFrameSetter*)framesetter, frameSize, &sizeOut, true);
+    _CTFrame* ret = _createFrame((_CTFramesetter*)framesetter, frameSize, &sizeOut, true);
+    if (ret->_lineOrigins.size() == 0) {
+        for (_CTLine* line in [ret->_lines objectEnumerator]) {
+            ret->_lineOrigins.emplace_back(line->_lineOrigin);
+        }
+    }
```

Comment:
Let us move this code to _createFrame. Ideally when a CTFrame object is created it should have all its members setup properly.